Objects/typevarobject.c

```diff
@@ -0,0 +1,1313 @@
+}
+
+/*[clinic input]
+typevartuple.__reduce__ as typevartuple_reduce
+
+[clinic start generated code]*/
+
+static PyObject *
+typevartuple_reduce_impl(typevartupleobject *self)
+/*[clinic end generated code: output=3215bc0477913d20 input=3018a4d66147e807]*/
+{
+    return PyUnicode_FromString(self->name);
+}
+
+static PyObject *
+typevartuple_mro_entries(PyObject *self, PyObject *args)
+{
+    PyErr_SetString(PyExc_TypeError,
+                    "Cannot subclass an instance of TypeVarTuple");
+    return NULL;
+}
+
+static PyMethodDef typevartuple_methods[] = {
+    TYPEVARTUPLE_TYPING_SUBST_METHODDEF
+    TYPEVARTUPLE_TYPING_PREPARE_SUBST_METHODDEF
+    TYPEVARTUPLE_REDUCE_METHODDEF
+    {"__mro_entries__", typevartuple_mro_entries, METH_O},
+    {0}
+};
+
+PyDoc_STRVAR(typevartuple_doc,
+"Type variable tuple.\n\
+\n\
+Usage:\n\
+\n\
+  Ts = TypeVarTuple('Ts')  # Can be given any name\n\
+\n\
+Just as a TypeVar (type variable) is a placeholder for a single type,\n\
+a TypeVarTuple is a placeholder for an *arbitrary* number of types. For\n\
+example, if we define a generic class using a TypeVarTuple:\n\
+\n\
+  class C(Generic[*Ts]): ...\n\
+\n\
+Then we can parameterize that class with an arbitrary number of type\n\
+arguments:\n\
+\n\
+  C[int]       # Fine\n\
+  C[int, str]  # Also fine\n\
+  C[()]        # Even this is fine\n\
+\n\
+For more details, see PEP 646.\n\
+\n\
+Note that only TypeVarTuples defined in global scope can be pickled.\n\
+");
+
+PyTypeObject _PyTypeVarTuple_Type = {
+    PyVarObject_HEAD_INIT(&PyType_Type, 0)
+    .tp_name = "typing.TypeVarTuple",
+    .tp_basicsize = sizeof(typevartupleobject),
+    .tp_dealloc = typevartupleobject_dealloc,
+    .tp_alloc = PyType_GenericAlloc,
+    .tp_iter = typevartupleobject_iter,
+    .tp_repr = typevartupleobject_repr,
+    .tp_flags = Py_TPFLAGS_DEFAULT | Py_TPFLAGS_IMMUTABLETYPE,
+    .tp_members = typevartuple_members,
+    .tp_methods = typevartuple_methods,
+    .tp_new = typevartuple,
+    .tp_doc = typevartuple_doc,
+};
+
+PyObject *_Py_make_typevar(const char *name, PyObject *bound_or_constraints) {
+    if (bound_or_constraints != NULL && PyTuple_CheckExact(bound_or_constraints)) {
+        return (PyObject *)typevarobject_alloc(name, NULL, bound_or_constraints, false, false, true);
+    } else {
+        return (PyObject *)typevarobject_alloc(name, bound_or_constraints, NULL, false, false, true);
+    }
+}
+
+PyObject *_Py_make_paramspec(const char *name) {
+    return (PyObject *)paramspecobject_alloc(name, NULL, false, false, true);
+}
+
+PyObject *_Py_make_typevartuple(const char *name) {
+    return (PyObject *)typevartupleobject_alloc(name);
+}
+
+static void
+typealias_dealloc(PyObject *self)
+{
+    typealiasobject *ta = (typealiasobject *)self;
+    free((void *)ta->name);
+    Py_XDECREF(ta->type_params);
+    Py_XDECREF(ta->compute_value);
+    Py_XDECREF(ta->value);
+    Py_TYPE(self)->tp_free(self);
```

Comment:
Crashes here. It seems `tp_free` is 0.

Reply:
I think I just fixed that. I am going to look at the other CI failures next.